src/librbd/api/Migration.cc

```diff
@@ -893,19 +1031,21 @@ int Migration<I>::abort() {
       return r;
     }
 
-    // copy dst HEAD -> src HEAD
-    SteppedProgressContext progress_ctx(m_prog_ctx, 2);
-    revert_data(m_dst_image_ctx, m_src_image_ctx, &progress_ctx);
-    progress_ctx.next_step();
+    SteppedProgressContext progress_ctx(
+      m_prog_ctx, (m_src_image_ctx != nullptr ? 2 : 1));
+    if (m_src_image_ctx != nullptr) {
```

Comment:
Just thinking (not related to this PR) it might be useful if we had an option not to copy dst HEAD to srt HEAD for "native" case when aborting, i.e. have an image as it was in the state when migration started.